plaidml/bridge/openvino/ops/reduce.cpp

```diff
@@ -19,71 +19,73 @@ void registerReduceOps() {
   registerOp("ReduceLogicalAnd", [](const Context& ctx) {
     IE_ASSERT(ctx.operands.size() == 2);
     auto I = ctx.operands.at(0);
-    std::vector<size_t> axes = get_axis_vector_from_constant_operand(1, ctx.layer);
+    std::vector<int64_t> axes = cast_constant_operand<int64_t>(1, ctx.layer);
     auto* layer = ngraph::as_type<ngraph::opset1::ReduceLogicalAnd>(ctx.layer);
+    I = edsl::cast(I, DType::FLOAT32);
```

Comment:
Because ReduceLogical Ops require bool precision in which NGraph will send `i1` tensor to plaidml, but it seems plaidml tensor `sum`(`LogicalOr`) and `product`(`LogicalAnd`) with `i1` datatype will get incorrect result. Here I simply cast input to `fp32` type.
I think you are right, maybe we can cast input tensor to `UINT8` as `all` and `any` ops originally want and cast back when returning output. Like this
```cpp
auto I_i8 = edsl::cast(I, DType::UINT8);
auto O = op::all(I_i8, edsl::make_tuple(axes), layer->get_keep_dims());
return edsl::make_tuple(edsl::cast(O, I.dtype()));
```

Reply:
Ok, makes sense.

I'll flag for the long term that there might be an additional problem in the oplib for op::all (and op::any) where the return type doesn't necessarily match the original type. OTOH, I think other frameworks might explicitly want the result of such an operation to be a different type the the inputs; I think it probably makes sense to leave oplib as-is, and cast back to the original type in the frontend (as you do here) where necessary.

So in summary, I like the approach you suggested in this comment.